### ISSUE TITLE
Repeatedly throwing "Unfinished session detected" while task is running

### ISSUE THREAD

**Issue by HWiese1980** [OpenDevin/OpenDevin#979]
#### Describe the bug
In the frontend I'm repeatedly and relatively frequently getting the modal dialog stating that an "unfinished session" has been "detected", offering me to load it or start a new session. Loading it replays every single chat message from the very beginning, which can  - depending on the task - take a while. Meanwhile the message pops up again. I do not want to start a new session. And I do not want to have to reload the existing session over and over again. 

#### Setup and configuration
**Current version**:

```bash
commit 707ab7b3f84fb5664ff63da0b52e7b0d2e4df545
```

**My config.toml and environment vars** (be sure to redact API keys):
```toml
LLM_MODEL="gpt-4"
LLM_API_KEY="<secret>"
LLM_BASE_URL="https://api.openai.com/v1/"
LLM_EMBEDDING_MODEL="openai"
WORKSPACE_DIR="./workspace"
```

**My model and agent** (you can see these settings in the UI):
* Model: openai/gpt-4
* Agent:

**Commands I ran to install and run OpenDevin**:
```
make build
make setup-config
make run
```

**Steps to Reproduce**:
Simply start a long running task. After a while the dialog should pop up. Unless something else causes this behavior... I do not know yet.

**Logs, error messages, and screenshots**:

**Comment by enyst** [comment]
A new session is supposed to mean only to not load the messages history. "Do not continue where you left off".

So maybe it is the option you want?

**Comment by enyst** [comment]
Ah. It might have a timer, or more than one session, not sure. Maybe it helps to delete the cache folder in opendevin directory.

**Comment by rbren** [comment]
This should only happen on page load...is it possible the page is refreshing for some reason?

**Comment by HWiese1980** [comment]
@rbren Maybe it's Chrome's energy saving mode that unloads a background tab until you open it again... I'll have to watch it.

**Comment by rbren** [comment]
I'll just note that I've definitely seen this message pop up more often than it should. I don't have specific repro steps available.

One weird flow though:
* Refresh page
* Tell it to start a new session
* Refresh again
* Get prompted again (should probably just be the fresh unstarted session?)
* Say "resume"
* You get back to the old original session

**Comment by enyst** [comment]
@HWiese1980 I believe this was solved. I'll close this, but please take a look at 0.4.0 and feel free to reopen to let us know if this still happens.

@rbren That scenario isn't happening anymore either. (On main, but I'm pretty sure the fixes were older than 0.4.0)
